src/CheckBox/index.js

```diff
@@ -17,41 +18,59 @@ import { Label, HiddenCheckbox, StyledCheckbox } from './elements';
 class CheckBox extends PureComponent {
   /** Internal state. */
   state = {
-    checked: this.props.checked, // eslint-disable-line
+    isChecked: this.props.isDefaultChecked, // eslint-disable-line
   };
 
   /**
-   * Component Did Update
-   * Method invoked immediately after updating occur.
+   * Returns true if requested prop exists
+   *
+   * @param {string} prop - requested property name
+   *
+   * @return {boolean}
    */
-  componentDidUpdate(prevProps) {
-    const { checked } = this.props;
-
-    // Update state only if props have changed
-    if (prevProps.checked !== checked) {
-      this.setState({
-        checked,
-      });
-    }
-  }
+  isControlled = prop => {
```

Comment:
@sun-tea What do you think about put that method and the `getControllableValue ` into an helper file to avoid repetition ?